### PR TITLE
simple mtcnn demo

### PR DESCRIPTION
**Pull request by vladimir-dudnik**
Simplified version of MTCNN Python demo, built on base of code from #1690 
The difference from original is that face reidentification functionality and images with unclear origin removed. To demonstrate face reidentification more convenient and probably interactive interface is desirable
The difference from #2133 is that this is much simpler (and limited, of course)

### CODE REVIEW

File: demos/tests/cases.py [hunk 1]
```diff
@@ -476,6 +476,18 @@ def single_option_cases(key, *args):
...
     )),
 
+    PythonDemo(name='face_detection_mtcnn_demo', device_keys=['-d'], test_cases=combine_cases(
+        TestCase(options={'--no_show': None,
+                          '-i': DataPatternArg('375x500')}),
+        [
+            TestCase(options={
+                '-m_p': ModelArg('mtcnn-p'),
+                '-m_r': ModelArg('mtcnn-r'),
+                '-m_0': ModelArg('mtcnn-o')
```
Comment (eizamaliev) [inline demos/tests/cases.py:486]:
```suggestion
                '-m_o': ModelArg('mtcnn-o')
```

**Review by eizamaliev — CHANGES_REQUESTED** [review]

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    rnet_input = []
+    for rectangle in rectangles:
+        crop_img = origin_image[int(rectangle[1]):int(rectangle[3]), int(rectangle[0]):int(rectangle[2])]
+        crop_img = image_reader(crop_img, 24, 24)
+        rnet_input.extend(crop_img)
+
+    rnet_res = exec_rnet.infer(inputs={rnet_input_blob: rnet_input})
+
+    (layer_name_roi, roi_prob), (layer_name_cls, cls_prob)  = rnet_res.items()
+    rectangles = tools.filter_face_24net(cls_prob, roi_prob, rectangles, ow, oh, score_threshold[1], iou_threshold[2])
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py:149]:
```suggestion
    # Start Rnet
    if len(rectangles) > 0:
        # Loading Rnet model to the plugin
        log.info("Loading Rnet model to the plugin")

        r_net.reshape({rnet_input_blob : [len(rectangles), 3, 24, 24]})  # Change batch size of input blob
        exec_rnet = ie_r.load_network(network = r_net, device_name = args.device)

        rnet_input = []
        for rectangle in rectangles:
            crop_img = origin_image[int(rectangle[1]):int(rectangle[3]), int(rectangle[0]):int(rectangle[2])]
            crop_img = image_reader(crop_img, 24, 24)
            rnet_input.extend(crop_img)

        rnet_res = exec_rnet.infer(inputs={rnet_input_blob: rnet_input})

        (layer_name_roi, roi_prob), (layer_name_cls, cls_prob)  = rnet_res.items()
        rectangles = tools.filter_face_24net(cls_prob, roi_prob, rectangles, ow, oh, score_threshold[1], iou_threshold[2])
```

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    onet_input = []
+    for rectangle in rectangles:
+        crop_img = origin_image[int(rectangle[1]):int(rectangle[3]), int(rectangle[0]):int(rectangle[2])]
+        crop_img = image_reader(crop_img, 48, 48)
+        onet_input.extend(crop_img)
+
+    onet_res = exec_onet.infer(inputs={onet_input_blob: onet_input})
+
+    (layer_name_roi, roi_prob), (layer_name_landmark, pts_prob), (layer_name_cls, cls_prob)  = onet_res.items()
+    rectangles = tools.filter_face_48net(cls_prob, roi_prob, pts_prob, rectangles, ow, oh, score_threshold[2], iou_threshold[3])
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py:172]:
```suggestion
    # Start Onet
    if len(rectangles) > 0:
        log.info("Loading Onet model to the plugin")

        o_net.reshape({onet_input_blob : [len(rectangles), 3, 48, 48]})  # Change batch size of input blob
        exec_onet = ie_o.load_network(network = o_net, device_name = args.device)

        onet_input = []
        for rectangle in rectangles:
            crop_img = origin_image[int(rectangle[1]):int(rectangle[3]), int(rectangle[0]):int(rectangle[2])]
            crop_img = image_reader(crop_img, 48, 48)
            onet_input.extend(crop_img)

        onet_res = exec_onet.infer(inputs={onet_input_blob: onet_input})

        (layer_name_roi, roi_prob), (layer_name_landmark, pts_prob), (layer_name_cls, cls_prob)  = onet_res.items()
        rectangles = tools.filter_face_48net(cls_prob, roi_prob, pts_prob, rectangles, ow, oh, score_threshold[2], iou_threshold[3])
```

**Review by eizamaliev — CHANGES_REQUESTED** [review]

File: demos/tests/cases.py [hunk 4]
```diff
@@ -476,6 +476,18 @@ def single_option_cases(key, *args):
        })
     )),
 
+    PythonDemo(name='face_detection_mtcnn_demo', device_keys=['-d'], test_cases=combine_cases(
+        TestCase(options={'--no_show': None,
+                          '-i': DataPatternArg('375x500')}),
```
Comment (eizamaliev) [inline demos/tests/cases.py:481]:
```suggestion
                          '-i': image_net_arg('00000002')}),
```

**Review by eizamaliev — CHANGES_REQUESTED** [review]

**Review by Wovchena — CHANGES_REQUESTED** [review]
Should https://github.com/openvinotoolkit/open_model_zoo/pull/2133 be closed?

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/face_detection_mtcnn_demo/python/tools_matrix.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+'''
+Function:
+    change rectangles into squares (matrix version)
+Input:
+    rectangles: rectangles[i][0:3] is the position, rectangles[i][4] is score
+Output:
+    squares: same as input
+'''
+def rect2square(rectangles):
```
[the remark this answers is not in the document]
Comment (eizamaliev) [reply demos/face_detection_mtcnn_demo/python/tools_matrix.py]: Done

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+        onet_input = []
+        for rectangle in rectangles:
+            crop_img = origin_image[int(rectangle[1]):int(rectangle[3]), int(rectangle[0]):int(rectangle[2])]
+            crop_img = image_reader(crop_img, 48, 48)
+            onet_input.extend(crop_img)
+
+        onet_res = exec_onet.infer(inputs={onet_input_blob: onet_input})
+
+        (layer_name_roi, roi_prob), (layer_name_landmark, pts_prob), (layer_name_cls, cls_prob)  = onet_res.items()
```
[the remark this answers is not in the document]
Comment (eizamaliev) [reply demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py]: Done

**Comment by eizamaliev** [comment]
Jenkins please retry a build

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+    args = build_argparser().parse_args()
+
+    # Plugin initialization for specified device and load extensions library if specified
+    log.info("Creating Inference Engine")
+
+    ie = IECore()
+
+    # Read IR
+    log.info("Loading network files:\n\t{}".format(args.model_pnet))
+    p_net = ie.read_network(args.model_pnet, args.model_pnet.with_suffix('.bin'))
```
Comment (Wovchena) [inline demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py:106]: Why did you update `read_network()` call with second arg?

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+            rectangle = utils.detect_face_12net(cls[0][1], roi[0], out_side, 1/scales[i], ow, oh, score_threshold[0], iou_threshold[0])
+            rectangles.extend(rectangle)
+        rectangles = utils.NMS(rectangles, iou_threshold[1], 'iou')
+
+        # Rnet stage
+        if len(rectangles) > 0:
+            log.info("Loading Rnet model to the plugin")
+
+            r_net.reshape({rnet_input_blob: [len(rectangles), 3, 24, 24]})  # Change batch size of input blob
+            exec_rnet = ie.load_network(network=r_net, device_name = args.device)
```
Comment (Wovchena) [inline demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py:217]:
```suggestion
            exec_rnet = ie.load_network(network=r_net, device_name=args.device)
```

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+            roi = rnet_res[rnet_roi_name]
+            cls = rnet_res[rnet_cls_name]
+            rectangles = utils.filter_face_24net(cls, roi, rectangles, ow, oh, score_threshold[1], iou_threshold[2])
+
+        # Onet stage
+        if len(rectangles) > 0:
+            log.info("Loading Onet model to the plugin")
+
+            o_net.reshape({onet_input_blob : [len(rectangles), 3, 48, 48]})  # Change batch size of input blob
+            exec_onet = ie.load_network(network = o_net, device_name = args.device)
```
Comment (Wovchena) [inline demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py:236]:
```suggestion
            exec_onet = ie.load_network(network=o_net, device_name=args.device)
```

File: demos/tests/cases.py [hunk 10]
```diff
@@ -476,6 +476,18 @@ def single_option_cases(key, *args):
        })
     )),
 
+    PythonDemo(name='face_detection_mtcnn_demo', device_keys=['-d'], test_cases=combine_cases(
+        TestCase(options={'--no_show': None,
+                          '-i': image_net_arg('00000002')}),
+        [
+            TestCase(options={
```
Comment (Wovchena) [inline demos/tests/cases.py:483]: This can be merged with previous `TestCase` instance.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py (hunk 7 above)
Comment (eizamaliev) [reply demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py]: It was made to support previous version of OV, but I realized, that it bad practice.

File: demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py (hunk 8 above)
Comment (eizamaliev) [reply demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py]: done

File: demos/tests/cases.py (hunk 10 above)
Comment (eizamaliev) [reply demos/tests/cases.py]: done

**Review by Wovchena — APPROVED** [review]

**Review by eizamaliev — APPROVED** [review]

**Pull request merged**